card/mockCard.go

```diff
@@ -758,6 +837,78 @@ func (c *MockCard) SendPhonons(keyIndices []uint16, extendedRequest bool) (trans
 // 	return nil
 // }
 
+func (c *MockCard) PostPhonons(recipientsPublicKey []byte, nonce uint64, keyIndices []uint16) (transferPhononPackets []byte, err error) {
+	log.Debug("sending mock POST_PHONONS command")
+	var outgoingPhonons []byte
+
+	for _, k := range keyIndices {
+		if int(k) >= len(c.Phonons) {
+			return nil, errors.New("keyIndex exceeds length of phonon list")
+		}
+		if c.Phonons[k].deleted {
+			return nil, errors.New("cannot access deleted phonon")
+		}
+
+		phonon := c.Phonons[k]
+
+		iv := util.RandomKey(16)
+
+		log.Debug("...encryptedPrivateKey")
+		encryptedPrivateKey, err := crypto.EncryptData(phonon.PrivateKey, recipientsPublicKey, iv)
+		if err != nil {
+			return nil, errors.New("could not encrypt private key")
+		}
+
+		log.Debug("...phononTLV")
+		phononTLV, err := phonon.EncodePosted(encryptedPrivateKey, iv)
+		if err != nil {
+			return nil, errors.New("could not encode phonon TLV")
+		}
+
+		outgoingPhonons = append(outgoingPhonons, phononTLV.Encode()...)
+	}
+
+	phononTransferTLV, err := tlv.NewTLV(TagTransferPhononPacket, outgoingPhonons)
+
+	if err != nil {
+		return nil, errors.New("could not encode phonon transfer TLV")
+	}
+
+	nonceBytes := make([]byte, 8)
+	binary.BigEndian.PutUint64(nonceBytes, uint64(nonce))
+
+	nonceTLV, err := tlv.NewTLV(TagNonce, nonceBytes)
+	if err != nil {
+		return nil, errors.New("could not encode nonce TLV")
+	}
+
+	cardCertTLV, err := tlv.NewTLV(TagCardCertificate, c.IdentityCert.Serialize())
+	if err != nil {
+		return nil, errors.New("could not encode cert TLV")
+	}
+
+	sig, err := ecdsa.SignASN1(rand.Reader, c.identityKey, CreatePostedPhononSignatureData(recipientsPublicKey, nonceBytes, outgoingPhonons))
+	if err != nil {
+		return nil, err
+	}
+
+	sigTLV, err := tlv.NewTLV(TagECDSASig, sig)
+	if err != nil {
+		return nil, errors.New("could not encode signedMessage TLV")
+	}
+
+	data := append(nonceTLV.Encode(), phononTransferTLV.Encode()...)
+	data = append(data, cardCertTLV.Encode()...)
+	data = append(data, sigTLV.Encode()...)
+
+	//Delete sent phonons
+	for _, k := range keyIndices {
+		c.deletePhonon(int(k))
+	}
+
+	return data, nil
+}
```

Comment:
This is really cool. I think this function looks good overall besides my nitpicks.